Realm.Shared/RealmObjectOps.cs

```diff
@@ -0,0 +1,393 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Realms
+{
+    [Preserve(AllMembers = true)]
+    internal static class RealmObjectOps
```

Comment:
I've modelled this class after [`System.Runtime.CompilerServices.RuntimeOps`](http://referencesource.microsoft.com/System.Core/Microsoft/Scripting/Actions/ExpandoObject.cs.html#35ec732587a0a4c3). The `DynamicMetaObject` for `System.Dynamic.ExpandoObject` emits expressions that call the methods on this class, much like the `MetaRealmObject` does here. I liked that pattern and I thought I'd copy it.

Another reason for this class to exist separately is that the methods here take a column index directly, because it can be cached in the dynamic callsite. It's also a cop-out, because the expression to look up the column index would be more complex. In any case, I hope that soon we start weaving the methods here and remove the protected methods on `RealmObject`.